TLM/TLM/Patch/_VehicleAI/_CarAI/StartPathFindPatch.cs

```diff
@@ -0,0 +1,42 @@
+namespace TrafficManager.Patch._VehicleAI._CarAI{
+    using HarmonyLib;
+    using JetBrains.Annotations;
+    using TrafficManager.Patch._PathManager;
+    using System.Reflection;
+    using UnityEngine;
+    using TrafficManager.State.ConfigData;
+    using CSUtil.Commons;
+    using TrafficManager.API.Traffic.Enums;
+    using TrafficManager.Manager.Impl;
+    using ColossalFramework;
+
+    [HarmonyPatch]
+    public class StartPathFindPatch {
+        public static MethodBase TargetMethod() => StartPathFindCommons.TargetMethod<CarAI>();
+
+        [UsedImplicitly]
+        public static void Prefix(ushort vehicleID, ref Vehicle vehicleData,
+            Vector3 startPos, Vector3 endPos, bool startBothWays, bool endBothWays, bool undergroundTarget) {
+#if DEBUG
+            // TODO [issue *]: why this message is not printed for all overrides of CarAI?
```

Comment:
I'll add to that comment: Why don't we use patching to do the debugs on the StartPathfind methods? That way we could keep the main methods super-clean.

Reply:
main method? you mean `CreatePathPatch()`
I was considering to move the debug over there so that I do not have repeated code on every `StartPathFind()`

Reply:
We can discuss this on #951. Ill reference the issue here.